include/exadg/utilities/spatial_aware_function.h

```diff
@@ -0,0 +1,62 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2024 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef EXADG_AERO_UTILITIES_SPATIAL_AWARE_FUNCTION_H_
+#define EXADG_AERO_UTILITIES_SPATIAL_AWARE_FUNCTION_H_
+
+#include <deal.II/base/function.h>
+
+namespace ExaDG
+{
+namespace Utilities
+{
+/**
+ * This is a @c dealii::Function, that provides information if its result is constant over space at
+ * every time.
+ */
+template<int dim, typename Number = double>
+class SpatialAwareFunction : public dealii::Function<dim, Number>
+{
+public:
+  using time_type = typename dealii::Function<dim, Number>::time_type;
+
+  SpatialAwareFunction(const unsigned int n_components = 1, const time_type initial_time = 0.0)
+    : dealii::Function<dim, Number>(n_components, initial_time)
+  {
+  }
+
+  /**
+   * Varies in space, i.e. is not constant over space at every time.
```

Comment:
It seems as if "at every time" is in conflict with the comment above "if the function varies in space for a given time" where you define the function in `FieldFunctions`.

To be more explicit, we  could add `double const time` to the interface of this function (and for the function `compute_time_factor()` below). I think this would make some of the comments superfluous.

Reply:
[ef52a6e](https://github.com/exadg/exadg/pull/644/commits/ef52a6ec89f956fff89d3ecea0b015c02f4e9a40)